packages/vega-schema/src/axis.js

```diff
@@ -3,128 +3,126 @@ import {
   numberValue, stringValue, booleanValue, colorValue,
   alignValue, baselineValue, fontWeightValue, dashArrayValue,
   booleanOrSignal, arrayOrSignal, numberOrSignal, stringOrSignal,
-  booleanOrNumberOrSignal
+  booleanOrNumberOrSignal,
+  def, enums, object, oneOf, orSignal, ref,
+  booleanType, numberType, stringType, signalRef
 } from './util';
 
-export default {
-  "refs": {
-    "labelOverlap": {
-      "oneOf": [
-        {"type": "boolean"},
-        {"enum": ["parity", "greedy"], "type": "string"},
-        {"$ref": "#/refs/signal"}
-      ]
-    },
-    "tickCount": {
-      "oneOf": [
-        {"type": "number"},
-        {"type": "string", "enum": timeIntervals},
-        {
-          "type": "object",
-          "properties": {
-            "interval": {
-              "oneOf": [
-                {"type": "string", "enum": timeIntervals},
-                {"$ref": "#/refs/signal"}
-              ]
-            },
-            "step": numberOrSignal
-          },
-          "required": ["interval"]
-        },
-        {"$ref": "#/refs/signal"}
-      ]
-    }
-  },
-  "defs": {
-    "axis": {
-      "type": "object",
-      "properties": {
-        "orient": {"enum": ["top", "bottom", "left", "right"]},
-        "scale": {"type": "string"},
-        "format": stringOrSignal,
-        "minExtent": numberValue,
-        "maxExtent": numberValue,
-        "offset": numberValue,
-        "position": numberValue,
-        "bandPosition": numberValue,
-        "values": arrayOrSignal,
-        "zindex": {"type": "number"},
+const overlapEnum = ['parity', 'greedy'];
+
+const labelOverlap = oneOf(
+  booleanType,
+  enums(overlapEnum),
+  signalRef
+);
+const labelOverlapRef = ref('labelOverlap');
+
+const tickCount = oneOf(
+  numberType,
+  enums(timeIntervals),
+  object({
+    _interval_: orSignal(enums(timeIntervals)),
+    step: numberOrSignal
+  }),
+  signalRef
+);
+const tickCountRef = ref('tickCount');
+
+const axisOrientEnum = [
+  'top',
+  'bottom',
+  'left',
+  'right'
+];
 
-        // TITLE CONFIG
-        "title": stringOrSignal,
-        "titlePadding": numberValue,
-        "titleAlign": alignValue,
-        "titleAngle": numberValue,
-        "titleX": numberValue,
-        "titleY": numberValue,
-        "titleBaseline": baselineValue,
-        "titleColor": colorValue,
-        "titleFont": stringValue,
-        "titleFontSize": numberValue,
-        "titleFontWeight": fontWeightValue,
-        "titleLimit": numberValue,
-        "titleOpacity": numberValue,
+const guideEncodeRef = def('guideEncode');
 
-        // DOMAIN CONFIG
-        "domain": {"type": "boolean"},
-        "domainColor": colorValue,
-        "domainOpacity": numberValue,
-        "domainWidth": numberValue,
+const axis = object({
+  _orient_: enums(axisOrientEnum),
+  _scale_: stringType,
```

Comment:
From what I understand here is that vega 5 will not support signals for the `scale` parameter within the axes properties? And based on the discussion in https://github.com/vega/vega/pull/1544 I understand there was an idea for a `transform` parameter that accept a signal within a scale, but I cannot find this in the code. Still hoping to be able to swap between log and linear scales through signals one day.

Reply:
Scales are still string-typed (no signals allowed), and after further discussion, scale types still determine the transform. We do have a potential path forward for supporting dynamic scale types (with some natural limitations), but that has not yet been implemented. The related feature requests remain open in the issue tracker.

Reply:
(Part of the reason for this is that there was some urgency to get Vega 5.0.0 released sooner to support a Vega-Lite 3 release.)

Reply:
Thanks for clarifying. It's unfortunate but understandble